tasks/main.yml

```diff
@@ -245,3 +245,6 @@
   when:
     - ansible_service_mgr == 'systemd'
     - not (keepalived_systemd_overrides | bool)
+
+- name: Flush handlers
```

Comment:
I don't think we need the flush handlers, it will be automatically done at playbook/role's end.

Can you remove it?

Reply:
It will run at playbook end, not role end. Which means that you can't assume that keepalived will be restarted in case there're some another roles are to run by the play.

While it might be ok in some cases, this assumption is not correct imo. So, for example, in OSA case we restart keepalived only when we trigger handlers flush in haproxy role [1]

[1] https://zuul.opendev.org/t/openstack/build/f325c519da894d4b98e943e8ca521634/log/job-output.txt#16182

Reply:
I believe the flush handlers should not be done in this role, as the role is "behaving" just fine. (This role is/was used outside OSA)
I am happy to add a meta: flush_handlers task in OSA in the playbook calling keepalived, after its role call. I think it's a better place.

If my memory serves well (I wrote this integration in OSA long ago!), I think the initial idea was to have one maintenance window, and therefore trigger all the handlers at one single point.

As you seem willing to restart keepalived before haproxy, I agree it can be changed by the introduction of a flush_handlers. This would be okay, as the downtime happening in case of an instance of keepalived restarting is close to 0 (ommitting misconfigurations). However, I would do that in OSA instead.

Reply:
Yeah, you;re right, we need to restart keepalived later as otherwise check scripts might go mad....